webapp/app/templates/lotse/form_familienstand.html

```diff
@@ -83,6 +88,10 @@
             }
         }
 
+        function remove_familienstand_date_input(){
+            delete_date_field_value('familienstand_date');
+        }
+
```

Comment:
I don't think we explicitly need to call this on document.ready()? And if I understand it correctly, we really shouldn't.
The value is set correctly in the field's definition when you request the site.
It is important that we delete the value when we hide a field - also on site request, yes. But that is already done in `show_hide_familienstand_date_field` - *only* if the field is hidden.

Reply:
You are correct, we should not delete the data on document.ready().

The main reason, I did not think about that is that we call it implicitly by calling `show_hide_all_fields`. Because while we call `show_hide_all_fields` in line 177 we also use it in line 179 for any change event of the radio buttons. And imo we need it there, because otherwise we do not change the marital status date (e.g. `married_since`) when the marital status has been changed.

But we could maybe just add:
`$(`input[type="radio"]`).change(remove_familienstand_date_input)` in line 178. WDYT?

Reply:
> And imo we need it there, because otherwise we do not change the marital status date (e.g. married_since) when the marital status has been changed.

But we already delete the marital status date there by calling `show_hide_familienstand_date_field()` and `show_hide_familienstand_married_lived_separated_since_field()` etc. in `show_hide_all_fields` for any change event of the radio buttons. So we actually don't have to add anything imo.

Reply:
I'll have another look

Reply:
Ok, it does not delete the data if you change from married to widowed. That makes sense because the field is also not hidden in this case.

Reply:
Just to make sure that we understand each other correctly: that means, we remove the function completely, yes?

Reply:
Let's call

Reply:
There was a misunderstanding about what data needs to be deleted. We've decided to delete all the dates whenever the marital status changes.